src/plugins/vis_type_vega/public/vega_visualization.test.js

```diff
@@ -0,0 +1,240 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import $ from 'jquery';
+
+import 'leaflet/dist/leaflet.js';
+import 'leaflet-vega';
+import { createVegaVisualization } from './vega_visualization';
+
+import vegaliteGraph from './__test__/vegalite_graph.json';
+import vegaGraph from './__test__/vega_graph.json';
+import vegaMapGraph from './__test__/vega_map_test.json';
+
+import { VegaParser } from './data_model/vega_parser';
+import { SearchAPI } from './data_model/search_api';
+
+import { createVegaTypeDefinition } from './vega_type';
+
+import {
+  setInjectedVars,
+  setData,
+  setSavedObjects,
+  setNotifications,
+  setKibanaMapFactory,
+} from './services';
+import { coreMock } from '../../../core/public/mocks';
+import { dataPluginMock } from '../../data/public/mocks';
+import { KibanaMap } from '../../maps_legacy/public/map/kibana_map';
+
+jest.mock('./default_spec', () => ({
+  getDefaultSpec: () => jest.requireActual('./__test__/default.spec.json'),
+}));
+
+jest.mock('./lib/vega', () => ({
+  vega: jest.requireActual('vega'),
+  vegaLite: jest.requireActual('vega-lite'),
+}));
+
+describe('VegaVisualizations', () => {
+  let domNode;
+  let VegaVisualization;
+  let vis;
+  let vegaVisualizationDependencies;
+  let vegaVisType;
+
+  let mockWidth;
+  let mockedWidthValue;
+  let mockHeight;
+  let mockedHeightValue;
+
+  const coreStart = coreMock.createStart();
+  const dataPluginStart = dataPluginMock.createStartContract();
+
+  const setupDOM = (width = 512, height = 512) => {
+    mockedWidthValue = width;
+    mockedHeightValue = height;
+    domNode = document.createElement('div');
+
+    mockWidth = jest.spyOn($.prototype, 'width').mockImplementation(() => mockedWidthValue);
+    mockHeight = jest.spyOn($.prototype, 'height').mockImplementation(() => mockedHeightValue);
+  };
+
+  setKibanaMapFactory((...args) => new KibanaMap(...args));
+  setInjectedVars({
+    emsTileLayerId: {},
+    enableExternalUrls: true,
+    esShardTimeout: 10000,
+  });
+  setData(dataPluginStart);
+  setSavedObjects(coreStart.savedObjects);
+  setNotifications(coreStart.notifications);
+
+  beforeEach(() => {
+    vegaVisualizationDependencies = {
+      core: {
+        uiSettings: coreStart.uiSettings,
+      },
+      plugins: {
+        data: {
+          query: {
+            timefilter: {
+              timefilter: {},
+            },
+          },
+        },
+      },
+    };
```

Comment:
According to TS `VegaVisualizationDependencies` seems that right implementation of this mock is:
```suggestion
    vegaVisualizationDependencies = {
      core: coreMock.createSetup(),
      plugins: {
        data: dataPluginMock.createSetupContract()
      },
    };
```